scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/rest/ScheduleRestApi.java

```diff
@@ -0,0 +1,42 @@
+package org.cloudfoundry.autoscaler.scheduler.rest;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.cloudfoundry.autoscaler.scheduler.rest.model.ApplicationScalingSchedules;
+import org.cloudfoundry.autoscaler.scheduler.service.ScalingScheduleManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Fujitsu
+ *
+ */
+@RestController
+@RequestMapping(value = "/v2/schedules")
+public class ScheduleRestApi {
+
+	@Autowired
+	ScalingScheduleManager scalingScheduleManager;
+
+	private Log logger = LogFactory.getLog(this.getClass());
+
+	@RequestMapping(method = RequestMethod.PUT)
+	public ResponseEntity<ApplicationScalingSchedules> createSchedule(
+			@RequestBody ApplicationScalingSchedules rawApplicationSchedules) {
+
+		logger.info("create schedule");
+		String appId = scalingScheduleManager.createSchedules(rawApplicationSchedules);
+		ApplicationScalingSchedules savedApplicationSchedules = scalingScheduleManager.getSchedules(appId);
+
+		return new ResponseEntity<>(savedApplicationSchedules, null, HttpStatus.CREATED);
```

Comment:
Second guessing timing windows is a waste of CPU. While anything is possible, just being nanoseconds off is the difference between what was sent and some other answer. What you are also saying is that the client needs to examine the response each and every time.
The fact that we did a PUT, either it was replaced or it wasn't. Anything in between just seems broken according to the PUT contract.